syserol/model.py

```diff
@@ -2,6 +2,7 @@
 import pandas as pd
 import numpy as np
 import tensorly as tl
+from functools import lru_cache
```

Comment:
You're only calling the function once, so remove `lru_cache`

Reply:
I re-added lru_cache because for the figure function, we call cross_validation once and then evaluate_diff, which calls it a second time and makes running time quite long. I'm wondering if it's better to just add the sum squared difference evaluation into the cross_validation function, just so that it doesn't have to call it again, since it is taking so long to run.